collector/filesystem_linux.go

```diff
@@ -46,9 +52,40 @@ func (c *filesystemCollector) GetStats() ([]filesystemStats, error) {
 			log.Debugf("Ignoring fs type: %s", labels.fsType)
 			continue
 		}
+		mutex.Lock()
+		if _, ok := stuckMounts[labels.mountPoint]; ok {
+			stats = append(stats, filesystemStats{
+				labels:      labels,
+				deviceError: 1,
+			})
+			log.Debugf("Mount point %q is in an unresponsive state", labels.mountPoint)
+			mutex.Unlock()
+			continue
+		}
+		mutex.Unlock()
+
+		// The success channel is use do tell the "watcher" that the stat
+		// finished successfully.The channel is closed on success.
+		success := make(chan struct{})
+		// Lock is used to ensure that a mount point isn't labelled as stuck
+		// even after success as there may be a race condition if a stat call
+		// finished at the same time the timeout procs.
```

Comment:
"at the same time the timeout procs"... hmm somehow that sentence fragment doesn't parse for me?